documentation/optee_design.md

```diff
@@ -648,7 +648,7 @@ specified by the GlobalPlatform TEE specifications.
 There are several types of user mode TAs, which differ by the way they are
 stored.
 
-#### "Normal" or Secure Storage Trusted Applications
+#### "Normal" (Bootstrap) or Secure Storage Trusted Applications
```

Comment:
Cenfusing, because previously the meaning was "Normal TAs are Secure Storage TAs". With the new section title, you are saying that normal TAs are boostrap TAs, and SS TAs are another kind. Then how should we understand the later section "Legacy or REE FS TAs"?

Reply:
That is what I'm trying to figure out myself and also clarify by doing this PR, i.e., so we and others understand what we are talking about and use the same naming when talking about it. We have invented a new thing but we haven't really described it somewhere. All our TA's has been updated to become a TA of type Bootstrap due to [this](https://github.com/OP-TEE/optee_os/blob/master/scripts/sign.py#L77). So after digging into this I thought that it must be that we have -
- "Legacy TA's", i.e., the ones using old format loaded from REE (before introducing the bootstrap header).
- "Bootstrap TA's" with new format, loaded from REE.
- "Secure Storage TA's", installable TA's coming from Bootstrap TA's (?).
- "Early TA's", TA's baked in to the TEE itself.

Unfortunately I didn't really participate in the reviews when introducing this, but if you can help me gather your thoughts around namings etc, I'll update the PR accordingly.

Reply:
@jbech-linaro 100% correct :+1: 

3rd bullet: indeed "Secure storage TAs" have to be in "bootstrap" format, while (4th bullet) "early TAs" may be in any format that can be loaded by the rpc mechanism i.e. legacy or bootstrap (it's just the loading from REE that is replaced by reading from a special section in the TEE binary).